CHANGELOG.next.toml

```diff
@@ -52,3 +52,40 @@ message = "Fix bug that can cause panics in paginators"
 references = ["smithy-rs#1903", "smithy-rs#1902"]
 meta = { "breaking" = false, "tada" = false, "bug" = true, "target" = "client"}
 author = "rcoh"
+
+[[aws-sdk-rust]]
+message = """
+The HTTP connector used when making requests is now configurable. This is especially useful when testing:
+
+```rust
+    let conn = TestConnection::new(vec![
+        // tuples of requests and responses
+    ]);
+    let conf = SdkConfig::builder()
+        .http_connector(conn.clone().into())
```

Comment:
Why does this require `clone()`?

Reply:
Because to look at the `TestConnection` after using it, you have to have your own copy. I'm going to remove this in favor of including a "real world" example the like of which you already proposed.

Reply:
I changed the example so this is no longer relevant